src/store/modules/aeternity.js

```diff
@@ -114,95 +97,125 @@ export default {
           },
         });
         commit('setSdk', { instance });
-        await dispatch('initTippingContractIfNeeded');
       }
     },
     async scanForWallets({ commit, state: { sdk } }) {
-      const scannerConnection = await BrowserWindowMessageConnection({
-        connectionInfo: { id: 'spy' },
-      });
-      const detector = await WalletDetector({ connection: scannerConnection });
+      let stopScan;
+      let resolve = null;
+      let rejected = (e) => {
+        throw e;
+      };
       // eslint-disable-next-line no-underscore-dangle
       const webWalletTimeout = this._vm.$isMobileDevice ? 0
         : setTimeout(() => commit('enableIframeWallet'), 10000);
 
-      return new Promise((resolve) => {
-        detector.scan(async ({ newWallet }) => {
-          if (!newWallet) return;
+      async function handleNewWallet({ newWallet }) {
+        try {
+          if (!newWallet) return null;
           clearInterval(webWalletTimeout);
           await sdk.connectToWallet(await newWallet.getConnection());
-          await sdk.subscribeAddress('subscribe', 'current');
-          const address = sdk.rpcClient.getCurrentAccount();
-          if (!address) return;
-          detector.stopScan();
+          const ret = await sdk.subscribeAddress('subscribe', 'current');
+          const { address: { current } } = ret;
+          const address = Object.keys(current)[0];
+          if (!address) return null;
+          if (stopScan) stopScan();
           commit('setAddress', address, { root: true });
-          resolve(address);
-        });
+          if (resolve) return resolve(address);
+          return address;
+        } catch (e) {
+          return rejected(e);
+        }
+      }
+      console.log('Scanning for wallets...');
+
+      const scannerConnection = new BrowserWindowMessageConnection({
+        connectionInfo: { id: 'spy' },
+      });
```

Comment:
```suggestion
      const scannerConnection = new BrowserWindowMessageConnection();
```
https://github.com/aeternity/aepp-sdk-js/blob/2743e4af828902d90e532447b6c526c833869e49/examples/browser/aepp/src/Connect.vue#L89-L91